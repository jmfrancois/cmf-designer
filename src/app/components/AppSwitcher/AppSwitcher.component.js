import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect, Inject } from '@talend/react-cmf';

class AppSwitcher extends React.Component {
	constructor(props) {
		super(props);
		this.onGenerateAppBtn = this.onGenerateAppBtn.bind(this);
	}

	onGenerateAppBtn() {
		this.props.dispatch({
			type: AppSwitcher.ACTION_TYPE_ADD_APP,
		});
	}

	render() {
		if (!this.props.cwd) {
			return (
				<div className="alert alert-danger">
					<p>Bad setup you don t have a working directory from the backend</p>
				</div>
			);
		}
		return (
			<div>
				<h2>Working directory</h2>
				<strong>{this.props.cwd}</strong>
				<div>
					{this.props.isWebapp ? null : (
						<button type="button" className="btn btn-primary" onClick={this.onGenerateAppBtn}>
							<Inject component="Icon" name="talend-plus" />
							Generate CMF Webapp
						</button>
					)}
				</div>
			</div>
		);
	}
}

AppSwitcher.ACTION_TYPE_ADD_APP = 'APP_SWITCHER_ADD_APP';
AppSwitcher.ACTION_TYPE_SET_CWD = 'APP_SWITCHER_SET_CWD';
AppSwitcher.displayName = 'AppSwitcher';
AppSwitcher.propTypes = {
	cwd: PropTypes.string.isRequired,
	isWebapp: PropTypes.bool,
	...cmfConnect.propTypes,
};

export default cmfConnect({})(AppSwitcher);
