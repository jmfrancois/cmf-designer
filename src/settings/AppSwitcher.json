{
  "props": {
    "AppSwitcher#default": {
      "saga": "handleApps",
      "appExpression": "getApp"
    }
  }
}
