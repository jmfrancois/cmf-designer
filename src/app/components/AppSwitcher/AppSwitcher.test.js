import React from 'react';
import { shallow } from 'enzyme';

import AppSwitcher from './AppSwitcher.component';

describe('AppSwitcher', () => {
	it('should render', () => {
		const wrapper = shallow(<AppSwitcher.WrappedComponent cwd="/Users/foo/github/test" />);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
