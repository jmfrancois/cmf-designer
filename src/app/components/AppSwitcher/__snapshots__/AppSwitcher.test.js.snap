// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSwitcher should render 1`] = `
<div>
  <h2>
    Working directory
  </h2>
  <strong>
    /Users/foo/github/test
  </strong>
  <div>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <Inject
        component="Icon"
        name="talend-plus"
      />
      Generate CMF Webapp
    </button>
  </div>
</div>
`;
